### PR TITLE
Adds check-multi-template and example check

### PR DESCRIPTION
**Pull request by absolutejam**
#### General

- [ ] Update Changelog following the conventions laid out on [Our CHANGELOG Guidelines ](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass

#### New Plugins

- [ ] Tests
  - **Pester tests are WIP.**

- [ ] Add the plugin to the README
  - **Not yet! Looking for input first.**

- [x] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose

I found myself reinventing the wheel when trying to check that certain element existed or were in a desired state on my Windows servers, so I wrote a check that was flexible and reusable. First I was copy & pasting it then find-replacing bits and pieces, so I decided to fully template the process. So far I've been able to replace 4 bespoke checks (and counting) with little effort :+1: 

It provides 2 levels of input (`CriticalItems`, `ImportantItems`) as I sometimes find it useful to define 'critical' and 'important' tier services. It also has some parameter parsing features which are outlined in the `-help` parameter. I mainly use these to pass multiple comma-separated strings into my check definitions when deploying via. SaltStack.

#### Known Compatibility Issues

~May be PowerShell 3+ (Can't remember if `Compare-Object` is). Will need to check and I'll add that to the checks if so.~

Tested and working with PowerShell 2.0 (Alterations in 6fe1dd60).

### CODE REVIEW

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

**Comment by majormoses** [comment]
I think I am clearly out of my depth on powershell here. Is there a way to maybe provide these a library functions to be consumed by other checks rather than copy, paste, modify? It seems like it is possible from my quick googling: https://blogs.technet.microsoft.com/heyscriptingguy/2010/08/10/how-to-reuse-windows-powershell-functions-in-scripts/

**Comment by absolutejam** [comment]
So in the commit, there's `check-multi-template.ps1` which is essentially the template and `check-adapters.ps1` which uses this template by importing it (dot sourcing):
```
# Import template
$here = Split-Path -Parent $MyInvocation.MyCommand.Path
. "$here\check-multi-template.ps1"
```

All that is left in this script is to define a `$CheckOptions` hashtable to provide things like the check name, messages for a 'critical' event and most importantly, two scriptblocks (anonymous functions) that provide most of the comparison logic. Then the script simply calls `Invoke-Main` to run it through the template. 

When comparing to Ruby and Python equivalents, this is probably on par with the amount of copying needed each time, but all of the boilerplate is hidden in each script that builds upon the template.

**Comment by majormoses** [comment]
@simonsteur @Seji64 as previous contributors any chance you guys could review this? My powershell foo is basically not worth mentioning.

File: check-multi-template/check-multi-template.ps1 [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+    Write-CheckResult
+
+}
+
+#ENDREGION Functions
+
+
+#REGION Main
+
+# Run 'Invoke-Mmain' if run directly, but don't if imported.
```
Comment (majormoses) [inline check-multi-template/check-multi-template.ps1]: `Invoke-Main`

**Comment by majormoses** [comment]
@absolutejam if you can confirm that everything has a changelog entry I am good to merge and release this.

**Comment by absolutejam** [comment]
I'm happy with it, but to compromise, I'll start writing some Pester tests and see if I can get AppVeyor set up to validate it. That way we can third pair of :robot: eyes on it.

**Comment by majormoses** [comment]
@absolutejam I saw that it was missing a changelog entry so I added it

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-windows/versions/2.3.0

**Pull request merged**